gno.land/cmd/gnoland/root.go

```diff
@@ -1,42 +1,59 @@
 package main
 
 import (
-	"context"
-	"fmt"
+	"flag"
 	"os"
 
+	tmcfg "github.com/gnolang/gno/tm2/pkg/bft/config"
 	"github.com/gnolang/gno/tm2/pkg/commands"
-	"github.com/peterbourgon/ff/v3"
-	"github.com/peterbourgon/ff/v3/fftoml"
+	"github.com/gnolang/gno/tm2/pkg/log"
 )
 
-func main() {
-	io := commands.NewDefaultIO()
-	cmd := newRootCmd(io)
+var logger = log.NewTMLogger(log.NewSyncWriter(os.Stdout))
 
-	if err := cmd.ParseAndRun(context.Background(), os.Args[1:]); err != nil {
-		_, _ = fmt.Fprintf(os.Stderr, "%+v\n", err)
-		os.Exit(1)
-	}
+type baseCfg struct {
+	rootDir  string
+	tmConfig tmcfg.Config
 }
 
-func newRootCmd(io *commands.IO) *commands.Command {
+func (bc *baseCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&bc.rootDir,
+		"root-dir",
+		"testdir",
+		"directory for config and data",
+	)
+}
+
+func newRootCmd() *commands.Command {
+	bc := baseCfg{}
+
 	cmd := commands.NewCommand(
 		commands.Metadata{
 			ShortUsage: "<subcommand> [flags] [<arg>...]",
-			ShortHelp:  "Starts the gnoland blockchain node",
-			Options: []ff.Option{
-				ff.WithConfigFileFlag("config"),
```

Comment:
I'm not sure what I prefer, but I think we should choose one of two extremes. Currently, we're in an intermediary step.

One extreme option is to clearly differentiate between config files and flags. For example, we can remove items like "persistentPeers" that are available both in the config file and as a start flag.

An alternative option is to establish a standard practice of using a dual approach (config file + flag). This could be implemented globally or exclusively in the init function.

To be honest, I'm not certain about a specific plan. I can only say that the current status is a hybrid mix of implicit and explicit approaches. It is inconsistent, with sometimes one way and sometimes two ways, depending on the commands.

You can refer to this issue on GitHub for more details: https://github.com/gnolang/gno/issues/731.